indico/modules/rb_new/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -90,9 +64,147 @@ class RoomEditModal extends React.Component {
         onClose();
     };
 
+    handleSubmit = async (formData) => {
+        const {actions: {updateRoom}, room: {id}} = this.props;
+        let rv = await updateRoom(id, formData);
+        if(rv.error){
+            return rv.error;
+        }
+    };
+
+    renderHeader = (title) => (
+        <Header key={title}>
+            <Translate>{title}</Translate>
+        </Header>
+    );
+
+    renderCheckboxField = (name, label) => (
+        <Field key={name}
+               name={name}
+               component={ReduxCheckboxField}
+               componentLabel={Translate.string(label)}
+               as={Checkbox} />
+    );
+
+    renderFormField = (name, label) => (
+        <Field key={name}
+               name={name}
+               component={ReduxFormField}
+               label={Translate.string(label)}
+               as="input" />
+    );
+
+    renderFieldList = (fields) => fields.map(({name, label}) => this.renderFormField(name, label));
+
+    renderCheckboxList = (checkboxes) => checkboxes.map(({name, label}) => this.renderCheckboxField(name, label));
+
+    renderFormGroup = (children, key) => (
+        <Form.Group widths="equal" key={key}>
+            {children}
+        </Form.Group>
+    );
+
+    renderGridColumn = (children, key) => (
+        <Grid.Column key={key}>
+            {children}
+        </Grid.Column>
+    );
+
+    formDetails = [
```

Comment:
I'd rather define a simple structure with just primitive values  (ie string, bools, etc) for data and then, when actually rendering the data, iterate over than and generate the JSX from it